File: C1.UWP.BarCode/VB/BarCodeSamples/StringToEnumConverter.vb
Imports C1.BarCode

Public Class StringToEnumConverter
    Implements IValueConverter
    Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        Return value.ToString()
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        Return CType([Enum].Parse(GetType(CodeType), value.ToString()), CodeType)
    End Function
End Class

File: C1.UWP.BarCode/CS/BarCodeSamples/StringToEnumConverter.cs
using C1.BarCode;
using System;
using Windows.UI.Xaml.Data;

namespace BarCodeSamples
{
    public class StringToEnumConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            return value.ToString();
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            return (CodeType)Enum.Parse(typeof(CodeType), value.ToString());
        }
    }
}
